extensions/typescript-language-features/src/features/semanticTokens.ts

```diff
@@ -0,0 +1,219 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { ITypeScriptServiceClient, ExecConfig, ServerResponse } from '../typescriptService';
+import * as Proto from '../protocol';
+
+export function register(selector: vscode.DocumentSelector, client: ITypeScriptServiceClient) {
+	const provider = new SemanticTokensProvider(client);
+	return vscode.languages.registerSemanticTokensProvider(selector, provider, provider.getLegend());
+}
+
+/**
+ * Prototype of a SemanticTokensProvider, relying on the experimental `encodedSemanticClassifications-full` request from the TypeScript server.
+ * As the results retured by the TypeScript server are limited, we also add a Typescript plugin (typescript-vscode-sh-plugin) to enrich the returned token.
+ */
+class SemanticTokensProvider implements vscode.SemanticTokensProvider {
```

Comment:
Is semantic highlighting gated behind a VS Code setting?

Reply:
setting added